### ISSUE TITLE
Support zipkin format instead/along jaeger

### ISSUE THREAD

**Issue by jcchavezs** [hypertrace/hypertrace-ingester#95]
Currently debugging and reproducing issues is very hard and problematic. If we need to understand why a span data turned into certain values, the most basic thing you do is try to reproduce the error. At the moment we don't have a way to do that, there isn't a way to obtain the original span that was ingested at the end of the pipeline. This is problematic because debugging is almost impossible.

The current approach to overcome this issue is to deploy a jaefer instance along with the hypertrace UI and download the data using the jaeger download button. While this feels like a solution it isn't because what you download isn't exactly what you ingested, formats are different: jaeger ingests over zipkin format (like other zipkin forks) or using thrift and neither of them is what you download from jaeger UI. To temporary overcome this issue I created this tool https://github.com/jcchavezs/jaeger2zipkin which allows you to convert jaeger downloaded trace into zipkin trace and then be able to reingest it but that isn't 100% reliable because there are two transformations in the middle.

What would truly embrace full debugability is that we can download the same format we ingest. For that we would need to support ingesting zipkin data (which is what most of our agents use for reporting) and also keep the raw payload in the messages along the kafka pipeline. We don't event need to understand zipkin format (e.g. we don't need to serialize/deserialize the raw payload on ever step in the pipeline) but just to store and serve in the API to download and reingest it.

Ping @kotharironak @tim-mwangi @JBAhire @rish691 @buchi-busireddy

**Comment by buchi-busireddy** [comment]
@jcchavezs Doesn't including the original span as a blob in every span increase the Kafka message size drastically?  Are you suggesting to enable that conditionally or always?
I thought if we finish the export feature that you logged earlier, we would be able to get the original span back when the trace is exported. Isn't it?

**Comment by jcchavezs** [comment]
Yes it might (around) duplicate the size of payload. Not sure how big is
that (it would be good to have numbers about the average size of a message
in the platform) but at least the increasement is deterministic.

Regarding the export function, the whole point of having the export
function is that we can access to the original message. If we translate
Jaeger -> enrichment -> Hypertrace -> Zipkin it will be kind of pointless
because when in a problem, we are using corrupted data to replicate a bug.

**Comment by buchi-busireddy** [comment]
I definitely like the idea and see the necessity. Just thinking of different ways to implement.
Thinking out loud, another major way to consider is storing the raw spans into a KV store and accessing them for the export feature. By retaining only the most recent spans and cap'ing the KV store, we can make sure we don't replicate the full data and at the same time, we don't increase the Kafka message size drastically. 
There could be other ideas too. I'd recommend a feature/design doc for this so that we can finalize on the approach and implement it.
what do you think?

**Comment by rish691** [comment]
So we ingest data in Zipkin format in the platform and convert it into RawSpan (our internal format) at entry. 

> To overcome this issue temporary I created this tool https://github.com/jcchavezs/jaeger2zipkin which allows you to convert jaeger downloaded trace into zipkin trace and then be able to re-ingest it but that isn't 100% reliable because there are two transformations in the middle.

What do we miss in this conversion? 
Exploring on these lines, can we add some metadata to RawSpan message and see if this conversion Zipkin <-> RawSpan can be made lossless? Then the download button in ui, just needs to invoke this converter. 

I can create a doc to further this discussion.
